setup.py

```diff
@@ -47,11 +42,19 @@
     include_package_data=True,
 
     # Allows `setup.py test` to work correctly with pytest
-    setup_requires=['numpy>=1.19.5'] + pytest_runner,
+    setup_requires=[] + pytest_runner,
```

Comment:
numpy isn't required for setup (I believe we need it in MDAnalysis core because we call numpy as part of our setup procedure)

Reply:
This is another thing I completely misunderstood. Maybe you could try to explain what is your criteria to determine what is required in the setup? If my repo depends fully on NumPy... how come it is not included in the setup_requires? 🤯  Does that mean that Python makes a difference between "What I need to install this repo" vs " what I need to install to make the code in the repo work"?

Reply:
So setuptools (which is the thing that's used to run these setup files) is a lot of magic that I haven't had the chance to full decipher, but yes it (tries to) discriminates between what's required to "setup the repo" and what's required to "run the code".

Essentially here is the entry for setup_requires: "A string or list of strings specifying what other distributions need to be present in order for the setup script to run. setuptools will attempt to obtain these (even going so far as to download them using EasyInstall) before processing the rest of the setup script or commands. "

My understanding is that most time you really don't need anything to run setup. In the MDA core library we need numpy because we compile some Cython code that requires numpy's header files during the setup process.